ecl/hql/hqlutil.cpp

```diff
@@ -9011,3 +9011,35 @@ IHqlExpression * queryTransformAssignValue(IHqlExpression * transform, IHqlExpre
         return value->queryChild(1);
     return NULL;
 }
+
+//-------------------------------------------------------------------------------------------------
+
+IHqlExpression * convertSetToExpression(bool isAll, size32_t len, const void * ptr, ITypeInfo * setType)
+{
+    HqlExprArray results;
+    const byte *presult = (const byte *) ptr;
+    const byte *presult_end = presult + len;
+
+    if(isUnicodeType(setType->queryChildType()))
+    {
+        while (presult < presult_end)
+        {
+            const size32_t numUChars = *((size32_t *) presult);
+            presult += sizeof(size32_t);
+            Owned<ITypeInfo> type = makeUnicodeType(numUChars,setType->queryChildType()->queryLocale());
+            results.append(*createConstant(createUnicodeValue(numUChars, presult, LINK(type))));
```

Comment:
minor:  You could pass elementType in here and delete the previous line.  The createUnicodeValue() function converts it to the correct type if unknown length.